tests/compatibility/SyncOperationsTestCase.test.nut

```diff
@@ -0,0 +1,97 @@
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+@include __PATH__+"/../Core.nut"
+
+// ReadOneSectorTestCase
+// Tests for SPIFlashLogger.read()
+class SyncOperationsTestCase extends Core {
+
+    logger = null;
+    _postfix = " - 09876543210987654321098765432109876543210987654321";
+    _max_sector_logs = 62;
+
+    function setUp() {
+        return Promise(function(resolve, reject) {
+            try {
+                if (!isAvailable()) {
+                    resolve();
+                    return;
+                }
+                local start = 0;
+                local end = start + 2;
+                start *= SPIFLASHLOGGER_SECTOR_SIZE;
+                end   *= SPIFLASHLOGGER_SECTOR_SIZE;
+                logger = SPIFlashLogger(start, end);
+                // Erase all data
+                logger.eraseAll(true);
+
+                for (local i = 0; i <= _max_sector_logs; i++) { // 175
+                    logger.write(i + _postfix);
+                    server.log(i);
+                }
+                resolve();
+            } catch (ex) {
+                reject("Unexpected error on test setup: " + ex);
+            }
+        }.bindenv(this));
+    }
+
+    function _checkFirstLastAndReadSync(vstart, vend, error) {
+        // Check first and last
+        assertEqual(vstart + _postfix, logger.first(), error + "Unexpected first log");
+        assertEqual(vend + _postfix, logger.last(), error + "Unexpected last log");
+        // minimal test of the readSync
+        assertEqual((1 + vstart) + _postfix, logger.readSync(2), error + "Unexpected 2-nd log value");
+        assertEqual((vend - 1) + _postfix, logger.readSync(-2), error + "Unexpected minus 2-nd log value");
+    }
+
+    function testReadSync() {
+        // Check first sector
+        assertEqual(logger.getPosition() < logger._start + SPIFLASHLOGGER_SECTOR_SIZE, true, "Wrong logger position");
+        // check reading in scope of one sector
+        _checkFirstLastAndReadSync(0, _max_sector_logs, "One sector test.");
+        // cross sector border: last object between two sectors
+        logger.write((_max_sector_logs + 1) + _postfix);
+        // Check that sector border was crossed
+        assertEqual(logger.getPosition() > logger._start + SPIFLASHLOGGER_SECTOR_SIZE, true);
+        // Check values after sector border crossing
+        _checkFirstLastAndReadSync(0, _max_sector_logs + 1, "Two sectors border test.");
+        // cross sector border
+        logger.write((_max_sector_logs + 2) + _postfix);
+        // last object in the second sector
```

Comment:
The same question here: why is there an assumption about the second sector?

Reply:
the second logger allocated on the same sectors.

Therefore it should recover write position.  And the last log is based on start code therefore it should be the same object